src/views/Home/DogDetail.js

```diff
@@ -0,0 +1,60 @@
+import { useEffect, useState } from 'react';
+import { useParams, Link, useHistory, useLocation } from 'react-router-dom';
+import { deleteDog, fetchDogId } from '../../services/dogs';
+
+export default function DogDetail() {
+
+  const { id } = useParams();
+  const [dog, setDog] = useState(null);
+  const [error, setError] = useState('');
+  const [loading, setLoading] = useState(true);
+  // const [success, setSuccess] = useState(false);
+  
+  const location = useLocation();
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const dogData = await fetchDogId(id);
+        setDog(dogData);
+        setLoading(false);
+        // setSuccess(true);
+      } catch (e) {
+        setError('Something went wrong, please try again.');
+      }
+    };
```

Comment:
Awesome work here Cole! Nice use of try/catch, error handling, and loading state!